src/main/java/be/syntra/devshop/DevshopFront/controllers/ProductController.java

```diff
@@ -50,22 +56,29 @@ public String displayProductOverview(Model model) {
     }
 
     @GetMapping("/details/{id}")
-    public String handleGet(@PathVariable Long id, Model model) {
-        Product product = productService.findById(id);
-        model.addAttribute("product", product);
-        model.addAttribute("cart", cartService.getCart());
-        return "product/productDetails";
+    public String handleGet(@PathVariable Long id, Model model, Principal user) {
```

Comment:
this method name bothers me every time i see it, i would suggest renaming it to 'getProductDetails'.... but that one is already taken...

Reply:
life is simple :) -> getProductDetailsById